server/app/api/speak.py

```diff
@@ -27,4 +28,51 @@ def get_reference():
     Returns:
         _type_: _description_
     """
-    return jsonify(dict(texts=["The cat sat on the mat.", ]))
\ No newline at end of file
+    with conn.cursor() as cursor:
+        cursor.execute('select * from reference')
+        result = cursor.fetchall()
+        return jsonify(result)
+
+@blueprint.route('/exists_user_id/<user_id>', methods=['GET'])
+def get_all_user_id_route(user_id):
+    with conn.cursor() as cursor:
+        cursor.execute('select exists (select distinct user_id from audio where user_id = %s)', user_id)
+        result = cursor.fetchall()
+        # [{"exists (select distinct user_id from audio where user_id = '<user_id'>": 1}], where value of 1 means true, 0 means false.
+        return jsonify(result)
+
+    
+@blueprint.route('/write_record', methods=['POST'])
+def write_record_route():
+    data = request.json
+    # data = {
+    #     "audio_id": 3,
+    #     "user_id": 1,
+    #     "url": "audio_id_user_id_ref_id.mp3",
+    #     "date": "2023-03-23 12:34:56",
+    #     "validated": True,
+    #     "ref_id": 5,
+    #     "sequence_matcher_score": 0.9,
+    #     "cer_score": 0.8,
+    #     "metaphone_match_score": 0.7
+    # }
+    try:
+        db_helper.write_record(data, conn)
+        return jsonify({"result": "success"})
+    except Exception as e:
+        return jsonify({"result": "error", "message": str(e)}), 400
+    
+@blueprint.route('/write_file/<url>', methods=['POST'])
+def write_file_route(url):
+    # data = request.json
+    data = "test_recording.mp3"
```

Comment:
Read uploaded file from request and upload to s3.

```
s3 = boto3.client('s3')
with open("FILE_NAME", "rb") as f:
    s3.upload_fileobj(f, "BUCKET_NAME", "OBJECT_NAME")
```

Reply:
Added